Examples/NoModules/Chapter 20/Ex20_17/Ex20_17.cpp

```diff
@@ -0,0 +1,20 @@
+// Range factories and range adaptors
+#include <iostream>
+#include <ranges>   // For views, range factories, and range adaptors
+
+using namespace std::ranges::views;
+
+bool isEven(int i) { return i % 2 == 0; }
+int squared(int i) { return i * i; }
+
+int main()
+{
+  for (int i : iota(1, 10))       // Lazily generate range [1,10)
```

Comment:
Can't verify with VC++, doesn't support iota yet

Reply:
Tip: you can use the ranges-v3 library.